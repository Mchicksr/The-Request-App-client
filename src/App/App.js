import React, {Component} from 'react';
import { Route } from 'react-router-dom'

import './App.css'
import config from '../config'
import Main from '../Main/Main'
import Sent from '../Sent/Sent'
import Form from '../Form/Form '
import Vote from '../Vote/Vote'
import Review from '../Review/Review'
import ReviewSent from '../Review/Review-Sent'
import SplashPage from '../splash/splash'
// import Store from '../Store/Store'

class App extends Component {
  state ={
    songs: []
    
  }
componentDidMount(){
  // console.log(config.API_ENDPOINT)
    fetch(`${config.API_ENDPOINT}/songs`)
  
  .then((res)=>{
    // console.log(res)
    if(!res.ok){
    return res.json().then(e=> Promise.reject(e))
    }
    return res.json()
})
.then((res)=>{
  console.log("check",res)
  this.setState({songs:res})
})
.catch(error=>{
console.log(error)
})
}


addSong(song) {
  
}

// ValidateSong(Name, value){
// if(value.trim() === ''){
//   return `${Name} is required`
// }
// return null
// }

//   validate(){
//   name: name => this.ValidateSong('name', name),

// }


  render(){
    console.log('TEST2', this.state.songs)
    console.log('init2')

  return (

    <main className='App'>
      
        <Route path ='/' component={Main} /> 
    
      <Route path='/S' component={SplashPage} />
      <Route path ='/Form' render={()=> <Form songs={this.state.songs}/> } />
     <Route path ='/sent' component={Sent} />
     <Route path ='/vote' render={()=>< Vote songs={this.state.songs}/>} />
     <Route path = '/Review' component={Review} />
     <Route path = '/Review-Sent' component={ReviewSent} />
      
    
    </main>
  );
  }
}

export default App;

//  {/* <a href = "../form/reqeust-form.html"></a> */}
//  <button class ="req">Send Request</button> 
//  {/* </a><br></br> */}
//  {/* <a href="../vote/vote.html"> */}
//    <button class ="req">Vote Best Choices</button> 
//  {/* </a><br></br> */}
//  {/* <a href="../review/review-form.html"> */}
// <button class ="req">Give a DJ Review</button> 
// {/* </a><br></br> */}

