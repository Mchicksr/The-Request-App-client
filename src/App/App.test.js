import React from 'react';
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react';
import App from './App';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

// import "@testing-library/jest-dom"



it('renders without crashing,',()=>{
const div = document.createElement('div');
ReactDOM.render( <BrowserRouter ><App /></BrowserRouter>, div)
})

it("matches snapshot 1", () => {
  // const tree = renderer.create(<Button label="click me please"></Button>).toJSON();
    const tree = renderer.create(  <a class="req" href="/Form"> Send Request </a>).toJSON();

  expect(tree).toMatchSnapshot();
})



//snapshot
//unit test on the form comp
//smoke test on view

//unit :test fields
  // test when it does not work
    //ex when the lenght is to long
    //test every if else in function and function

