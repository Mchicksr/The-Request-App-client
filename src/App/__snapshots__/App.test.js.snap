// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<a
  class="req"
  href="/Form"
>
   Send Request 
</a>
`;
